tensorflow/python/ops/array_ops.py

```diff
@@ -1757,6 +1757,176 @@ def _DepthToSpaceShape(op):
       [input_shape[0], height, width, new_depth])]
 
 
+def one_hot(indices, depth, on_value=None, off_value=None,
+            axis=None, dtype=None, name=None):
+  """Returns a one-hot tensor.
+
+  The locations represented by indices in `indices` take value `on_value`,
+  while all other locations take value `off_value`. By default, `on_value` is 1,
+  and `off_value` is 0.
+
+  The parameter `dtype` specifies the type of the output tensor. If neither
+  `on_value` or `off_value` are specified, then they will take on the values of 
+  1 and 0, respectively, with the type of `dtype`. If `on_value`, `off_value`, 
+  and `dtype` are all provided, then `one_hot()` will check to ensure that
+  `on_value` and `off_value` match the given type. If none of these parameters 
+  are provided, then `on_value` and `off_value` will take on the values 1 and 0,
+  respectively, with the type `tf.float32`
+
+  If the input `indices` is rank `N`, the output will have rank `N+1`. The
+  new axis is created at dimension `axis` (default: the new axis is appended
+  at the end).
+
+  If `indices` is a scalar the output shape will be a vector of length `depth`
+
+  If `indices` is a vector of length `features`, the output shape will be:
+  ```
+    features x depth if axis == -1
+    depth x features if axis == 0
+  ```
+
+  If `indices` is a matrix (batch) with shape `[batch, features]`, the output
+  shape will be:
+  ```
+    batch x features x depth if axis == -1
+    batch x depth x features if axis == 1
+    depth x batch x features if axis == 0
+  ```
+
+
+  Examples
+  =========
+
+  Suppose that
+
+  ```
+    indices = [0, 2, -1, 1]
+    depth = 3
+    on_value = 5.0
+    off_value = 0.0
+    axis = -1
+  ```
+
+  Then output is `[4 x 3]`:
+
+  ```output =
+    [5.0 0.0 0.0]  // one_hot(0)
+    [0.0 0.0 5.0]  // one_hot(2)
+    [0.0 0.0 0.0]  // one_hot(-1)
+    [0.0 5.0 0.0]  // one_hot(1)
+  ```
+
+  Suppose that
+
+  ```
+    indices = [0, 2, -1, 1]
+    depth = 3
+    on_value = 0.0
+    off_value = 3.0
+    axis = 0
+  ```
+
+  Then output is `[3 x 4]`:
+
+  ```
+    output =
+    [0.0 3.0 3.0 3.0]
+    [3.0 3.0 3.0 0.0]
+    [3.0 3.0 3.0 3.0]
+    [3.0 0.0 3.0 3.0]
+  //  ^                one_hot(0)
+  //      ^            one_hot(2)
+  //          ^        one_hot(-1)
+  //              ^    one_hot(1)
```

Comment:
I copied these examples from the original C++ documentation- I believe the second example is showcasing the difference between `axis = 0` and `axis = -1`. I'll still take it out if you think that's best.